### PR TITLE
feat: add stepViewEvent to step block

### PR DESCRIPTION
**Pull request by csiyang**
# Description

Creates a stepViewEvent every time a step block is viewed on a journey

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/26787374/todos/4816009972)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] stepViewEventCreate mutation should be called once when you view a step
- [ ] stepViewEventCreate mutation should not be called in admin

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

**Comment by csiyang** [comment]
This PR requires [this](https://github.com/JesusFilm/core/pull/566) to be merged first

File: libs/journeys/ui/src/components/Step/Step.tsx [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
 interface StepProps extends TreeBlock<StepFields> {
+  uuid?: () => string
   wrappers?: WrappersProps
 }
 
-export function Step({ children, wrappers }: StepProps): ReactElement {
+export function Step({
+  id: blockId,
+  children,
+  uuid = uuidv4,
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Step/Step.tsx:25]: this is just for testing. You should instead stub this in jest. @GeronimoJohn had to learn how to do this. Maybe follow up with him how to do it.

File: libs/journeys/ui/src/components/Step/Step.tsx [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+
+  async function createResponse(blockId: string): Promise<void> {
+    const id = uuid()
+    await stepViewEventCreate({
+      variables: { input: { id, blockId } }
+    })
+  }
+
+  useEffect(() => {
+    if (wrappers == null) {
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Step/Step.tsx:40]: This is not a good metric for determining if this is contained on journeys.

File: libs/journeys/ui/src/components/Step/Step.tsx [hunk 3]
```diff
@@ -1,13 +1,48 @@
...
+      void createResponse(blockId)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Step/Step.tsx:43]: should not just disable this check.

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**